single_tech_samples/synapseanalytics/sample3_integration_testing/scripts/deploy_infrastructure.sh

```diff
@@ -0,0 +1,282 @@
+
+# Get endpoints for Azure resources
+azcloud=$(az cloud show --output json)
+storageEndpoint=$(echo "$azcloud" | jq -r '.suffixes.storageEndpoint')
+keyVaultEndpoint=$(echo "$azcloud" | jq -r '.suffixes.keyvaultDns')
+managementEndpoint=$(echo "$azcloud" | jq -r '.endpoints.resourceManager')
+synapseEndpoint=$(echo "$azcloud" | jq -r '.endpoints.synapseAnalyticsResourceId')
+
+# Validate arm template
+
+echo "Validating deployment"
+arm_output=$(az deployment group validate \
+    --resource-group "$resource_group_name" \
+    --template-file "./infrastructure/main.bicep" \
+    --parameters @"./infrastructure/main.parameters.${ENV_NAME}.json" \
+    --parameters project="${PROJECT}" keyvault_owner_object_id="${kv_owner_object_id}" deployment_id="${DEPLOYMENT_ID}" synapse_sqlpool_admin_password="${SYNAPSE_SQL_PASSWORD}" \
+    --output json)
+
+# Deploy arm template
+echo "Deploying resources into $resource_group_name"
+arm_output=$(az deployment group create \
+    --resource-group "$resource_group_name" \
+    --template-file "./infrastructure/main.bicep" \
+    --parameters @"./infrastructure/main.parameters.${ENV_NAME}.json" \
+    --parameters project="${PROJECT}" deployment_id="${DEPLOYMENT_ID}" keyvault_owner_object_id="${kv_owner_object_id}" synapse_sqlpool_admin_password="${SYNAPSE_SQL_PASSWORD}" \
+    --output json)
+
+if [[ -z $arm_output ]]; then
+    echo >&2 "ARM deployment failed."
+    exit 1
+fi
+
+
+########################
+# RETRIEVE KEYVAULT INFORMATION
+
+echo "Retrieving KeyVault information from the deployment."
+
+kv_name=$(echo "$arm_output" | jq -r '.properties.outputs.keyvault_name.value')
+kv_dns_name="https://${kv_name}${keyVaultEndpoint}/"
+
+
+# Store in KeyVault
+az keyvault secret set --vault-name "$kv_name" --name "kvUrl" --value "$kv_dns_name"
+az keyvault secret set --vault-name "$kv_name" --name "subscriptionId" --value "$AZURE_SUBSCRIPTION_ID"
+
+
+#########################
+# CREATE AND CONFIGURE SERVICE PRINCIPAL FOR ADLS GEN2
+
+# Retrive account and key
+azure_storage_account=$(echo "$arm_output" | jq -r '.properties.outputs.storage_account_name.value')
+azure_storage_key=$(az storage account keys list \
+    --account-name "$azure_storage_account" \
+    --resource-group "$resource_group_name" \
+    --query [0].value --output tsv)
+
+# Add file system storage account
+storage_file_system=datalake
+echo "Creating ADLS Gen2 File system: $storage_file_system"
+az storage container create --name $storage_file_system --account-name "$azure_storage_account" --account-key "$azure_storage_key"
+
+processed_file_system=saveddata
+az storage container create --name $processed_file_system --account-name "$azure_storage_account" --account-key "$azure_storage_key"
+
+echo "Creating folders within the file system."
+# Create folders for databricks libs
+az storage fs directory create -n '/sys/databricks/libs' -f $storage_file_system --account-name "$azure_storage_account" --account-key "$azure_storage_key"
```

Comment:
I don't think we necessarily need all these folders. We prob just need a landing folder and an output folder

Reply:
Good call out. I removed the unused containers.